pkg/backup/backupschedule/backup_schedule_manager.go

```diff
@@ -479,6 +479,10 @@ func caculateExpiredBackups(backupsList []*v1alpha1.Backup, reservedTime time.Du
 	expiredTS := config.TSToTSO(time.Now().Add(-1 * reservedTime).Unix())
 	i := 0
 	for ; i < len(backupsList); i++ {
+		// the backup status CommitTs will be empty after created. without this, all newly created backups will be GC'ed
+		if len(backupsList[i].Status.CommitTs) == 0 && (v1alpha1.IsBackupScheduled(backupsList[i]) || v1alpha1.IsBackupRunning(backupsList[i]) || v1alpha1.IsBackupPrepared(backupsList[i])) {
```

Comment:
We filter out newly created backups at line-358 instead.

Reply:
updated @grovecai